ext/native/net/http_client.cpp

```diff
@@ -562,6 +565,7 @@ void Downloader::Update() {
 	for (size_t i = 0; i < downloads_.size(); i++) {
 		if (downloads_[i]->Progress() == 1.0f || downloads_[i]->Failed()) {
 			downloads_[i]->RunCallback();
+			downloads_[i]->Join();
```

Comment:
We need to do this in CancelAll() too, or we'll crash now.  For perf, I'd say a separate loop joining them after they are all marked canceled (though, not sure we will often have multiple downloads.)

-[Unknown]

Reply:
Good catch! I'll fix it up.